A/abPOA/build_tarballs.jl

```diff
@@ -0,0 +1,61 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message
+using BinaryBuilder, Pkg
+
+name = "abPOA"
+version = v"1.4.1"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource(
+        "https://github.com/yangao07/abPOA.git",
+        "c6c9dacf92414cd1a358cbd1d28b6f2ca37b30ed",
+    ),
+    DirectorySource("./bundled"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/abPOA
+install_license LICENSE
+atomic_patch -p1 ../patches/no_march_native.patch
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+      -DCMAKE_BUILD_TYPE=Release \
+      -DBUILD_SHARED_LIBS=ON ..
+cmake --build . --parallel ${nproc} --target all
+cmake --build . --target install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter(p -> arch(p) == "x86_64", supported_platforms(; exclude = Sys.iswindows))
+
+# The products that we will ensure are always built
+products = [ExecutableProduct("abpoa", :abpoa), LibraryProduct("libabpoa", :libabpoa)]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    BuildDependency("SIMDe_jll"),
+    Dependency(
+        PackageSpec(; name = "Zlib_jll", uuid = "83775a58-1f1d-513f-b197-d71354ab007a");
+        compat = "1.2.13",
```

Comment:
Is there a reason you need this tight a version bound on zlib?

Reply:
![image](https://github.com/JuliaPackaging/Yggdrasil/assets/91387710/7399d15a-fe92-461b-9c16-e2ae19123816)
It's better to have a tighter compat bound than no bound as it might happen that another contributor might try to update this build recipe with a Zlib 2.x that breaks compatibility. Whenever we bump to a next version we usually try to update all of the dependencies anyway.

Reply:
I don't think that really answers the question. Restricting zlib version by extension means restricting Julia versions. I believe it'd be better to relax this compat at least what was used in Julia v1.6

Reply:
I see. In this case, based on https://github.com/JuliaPackaging/Yggdrasil/pull/2031, should I set compat to v1.2.12?

Reply:
To determine the version of stdlibs in Julia v1.6.0 I usually look at their Project.toml file: https://github.com/JuliaLang/julia/blob/f9720dc2ebd6cd9e3086365f281e62506444ef37/stdlib/Zlib_jll/Project.toml#L3

Reply:
That's good advice, thank you! Looks like I can set compat to 1.2.12